### ISSUE TITLE
"rustup self uninstall" fails on EndeavourOS.

### ISSUE THREAD

**Issue by jessezwd** [rust-lang/rustup#4135]
### Verification

- [x] I searched for recent similar issues at https://github.com/rust-lang/rustup/issues?q=is%3Aissue+is%3Aopen%2Cclosed and found no duplicates.
- [x] I am on the latest version of Rustup according to https://github.com/rust-lang/rustup/tags and am still able to reproduce my issue.

### Problem

$ rustup self uninstall

Thanks for hacking in Rust!

This will uninstall all Rust toolchains and data, and remove
$HOME/.cargo/bin from your PATH environment variable.

Continue? (y/N) 

info: aborting uninstallation

### Steps

rustup self uninstall

### Rustup version

```console
rustup 1.27.1 (54dd3d00f 2024-04-24):
```

### Installed toolchains

```console
installed targets for active toolchain
--------------------------------------

wasm32-unknown-unknown
x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.83.0 (90b35a623 2024-11-26)
```

### OS version

```console
EndeavourOS rolling version
```

**Comment by djc** [comment]
Can you try if this works better with https://internals.rust-lang.org/t/seeking-beta-testers-for-rustup-v1-28-0/22060? I'm not familiar with EndeavourOS, what is it and how is it different from other Linux distributions?

**Comment by rami3l** [comment]
> Can you try if this works better with [internals.rust-lang.org/t/seeking-beta-testers-for-rustup-v1-28-0/22060](https://internals.rust-lang.org/t/seeking-beta-testers-for-rustup-v1-28-0/22060)? I'm not familiar with EndeavourOS, what is it and how is it different from other Linux distributions?

@djc Just checked, Endeavour is Arch-based. If, say, `rustup` is installed via `pacman` then this will be difficult...

**Comment by jessezwd** [comment]
$ pacman -Q rustup
error: package 'rustup' was not found

**Comment by rami3l** [comment]
@jessezwd Do you happen to have a minimal reproduction on a reproducible environment, say, Arch's official docker image, so that it might be easier to investigate what have happened?

**Comment by jessezwd** [comment]
Thanks for your reply. 
I don't think so, no docker on my Computer. I might install rustup via pacman/yay. 
if I manually
1. remove .cargo
2. remove .rustup
3. delete the line: . "$HOME/.cargo/env" on .bashrc/.bash_profile
Would that be Ok?

**Comment by rami3l** [comment]
@jessezwd Yes, that would be enough, assuming you don't have other stuff like `zsh` or `fish` installed.

**Comment by jessezwd** [comment]
Thanks a lot!
